File: opcua-webservice-spring/core/src/main/java/org/opcfoundation/webserver/addressspace/nodes/UaReferenceType.java
package org.opcfoundation.webserver.addressspace.nodes;

import org.eclipse.milo.opcua.stack.core.types.builtin.LocalizedText;
import org.eclipse.milo.opcua.stack.core.types.builtin.NodeId;
import org.eclipse.milo.opcua.stack.core.types.enumerated.NodeClass;
import org.springframework.lang.Nullable;

public class UaReferenceType extends UaDefinitionNode {
    private final LocalizedText inverseName;
    private final boolean symmetric;

    public UaReferenceType(
            NodeId nodeId,
            String browseName,
            LocalizedText displayName,
            boolean isAbstract,
            LocalizedText inverseName,
            boolean symmetric)
    {
        super(nodeId,browseName,displayName,isAbstract);
        this.inverseName = inverseName;
        this.symmetric = symmetric;
    }

    @Override
    public NodeClass nodeClass() {
        return NodeClass.ReferenceType;
    }

    public LocalizedText inverseName() {
        return (symmetric) ? displayName : inverseName;
    }

    public boolean isSymmetric() {
        return symmetric;
    }

    public void setParentReferenceType(UaReferenceType referenceType)
    {
        setParentType(referenceType);
    }
}

File: opcua-webclient-ts/src/common/nodes/UaReferenceType.ts
import { NodeClass } from "opcua-webapi";
import { UaLocalizedText, UaNodeId } from "../types";
import { UaDefintionNode } from ".";

export class UaReferenceType extends UaDefintionNode
{    
    private _inverseName: UaLocalizedText;
    private _symmetric: boolean;
    
    constructor(
        nodeId: UaNodeId,
        browseName: string,
        displayName: UaLocalizedText,
        isAbstract: boolean,
        inverseName: UaLocalizedText,
        symmetric: boolean)
    {
        super(nodeId, browseName, displayName, isAbstract);

        this._inverseName = inverseName;
        this._symmetric = symmetric;
    }
    
    get nodeClass() : NodeClass 
    {
        return NodeClass.ReferenceType;
    }

    get inverseName() : UaLocalizedText
    {
        return this._inverseName;
    }

    get symmetric() : boolean 
    {
        return this._symmetric;
    }
}